includes/class-newspack-popups-segmentation.php

```diff
@@ -151,47 +123,59 @@ public static function register_custom_dimensions( $default_dimensions ) {
 		return $default_dimensions;
 	}
 
-
-	/**
-	 * Get GA property ID from Site Kit's options.
-	 */
-	public static function get_ga_property_id() {
-		$ga_settings = get_option( 'googlesitekit_analytics_settings' );
-		if ( $ga_settings && isset( $ga_settings['propertyID'] ) ) {
-			return $ga_settings['propertyID'];
-		}
-	}
-
 	/**
-	 * Inset GTAG amp-analytics with a remote config, which will insert segmentation-related custom dimensions.
+	 * Add custom custom dimensions to Newspack Plugin's Analytics reporting.
+	 *
+	 * @param array $custom_dimensions_values Existing custom dimensions payload.
 	 */
-	public static function insert_gtag_amp_analytics() {
-
+	public static function report_custom_dimensions( $custom_dimensions_values ) {
 		$custom_dimensions = [];
 		if ( class_exists( 'Newspack\Analytics_Wizard' ) ) {
 			$custom_dimensions = Newspack\Analytics_Wizard::list_configured_custom_dimensions();
 		}
-		$custom_dimensions_existing_values = [];
-		if ( class_exists( 'Newspack\Analytics' ) ) {
-			$custom_dimensions_existing_values = Newspack\Analytics::get_custom_dimensions_values( get_the_ID() );
+		if ( empty( $custom_dimensions ) ) {
+			return $custom_dimensions_values;
+		}
+
+		$client_id           = self::get_client_id();
+		$api                 = self::load_lightweight_api();
+		$subscription_events = $api->get_reader_events( $client_id, 'subscription' );
+		$donation_events     = $api->get_reader_events( $client_id, 'donation' );
+
+		foreach ( $custom_dimensions as $custom_dimension ) {
+			// Strip the `ga:` prefix from gaID.
+			$dimension_id = substr( $custom_dimension['gaID'], 3 ); // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+			switch ( $custom_dimension['role'] ) {
+				case Segmentation::CUSTOM_DIMENSIONS_OPTION_NAME_READER_FREQUENCY:
+					$read_count = Campaign_Data_Utils::get_post_view_count( $api->get_reader( $client_id ) );
+					// Tiers mimick NCI's – https://news-consumer-insights.appspot.com.
+					$read_count_tier = 'casual';
```

Comment:
This value doesn't seem to change in my testing—it always says `casual` no matter how many posts I view. I verified that the reader row in the custom table contains multiple post views:

```
+----------------------------+---------------------+---------------------+---------------------------------------------------------------------+------------+
| client_id                  | date_created        | date_modified       | reader_data                                                         | is_preview |
+----------------------------+---------------------+---------------------+---------------------------------------------------------------------+------------+
| amp-O3PbkPCQrtThfyrvy9xhqg | 2022-11-22 22:22:22 | 2022-11-22 22:25:56 | {"views":{"page":2,"post":8},"category":{"4":4,"3":4,"2":4,"12":4}} |       NULL |
+----------------------------+---------------------+---------------------+---------------------------------------------------------------------+------------+
```

Reply:
Thanks for catching that! The recent changes to client reconciliation changed the signature of `Campaign_Data_Utils::get_post_view_count`. Fixed in b59ab125003d39f6988548e91cc37de2e8f56367